### ISSUE TITLE
Duplicate aria handling for TimeNav

### ISSUE THREAD

**Issue by JoeGermuska** [NUKnightLab/TimelineJS3#791]
If i'm reading this correctly, the aria attributes set in `Timeline.js`

https://github.com/NUKnightLab/TimelineJS3/blob/accessibility-i18n/src/js/timeline/Timeline.js/#L432-L433

 and `TimeNav.js`

https://github.com/NUKnightLab/TimelineJS3/blob/accessibility-i18n/src/js/timenav/TimeNav.js/#L46-L47

overlap; however, they have different opinions about the value for `role` (`Timeline.js` says `group` but `TimeNav.js` says `application`).

Would love input from @oleksandr-danylchenko about whether I'm reading this correctly, and if so, how we might express a rationale for where to put things so that they don't get duplicated.  I think my preference is for "inside" -- that is, `TimeNav.js` should handle it and `Timeline.js` should not meddle in other classes -- but I have less practice with ARIA annotation.

(This came up in work on #780 )

**Comment by oleksandr-danylchenko** [comment]
@JoeGermuska 
Good catch :raised_hands: 
I also consider this as duplication and would rather **put the label definition into the `TimeNav.js`**. Because in that way the `TimeNav` encapsulates its a11y labels handling. And I cannot make up a strong reason on why the `Timeline.js` should set the label :thinking:

**Comment by oleksandr-danylchenko** [comment]
But I'm a bit worried if the `_` shouldn't be used to access the i18n methods. Because it's not that intuitive and has a strong association with the "private" properties (before the actual private props were introduced in JS)

**Comment by oleksandr-danylchenko** [comment]
But I'm concerned about the fact that we use the `_` as the way to access the i18n methods :thinking:. 
This symbol meaning isn't intuitive and has a strong association with the "private" properties (before the actual private props were introduced in JS)

**Comment by JoeGermuska** [comment]
@oleksandr-danylchenko thanks for weighing in. I'll make the change ASAP.

Regarding the use of `_`, that's directly inspired by Python's [`gettext`](https://docs.python.org/3/library/gettext.html), which was my primary experience with I18N before cooking up something for TimelineJS. The philosophy, as I've inferred, is that many apps will have enough I18N that a longer function name becomes cluttery.

That's something we could discuss, but I'd prefer to leave it out of the Accessibility 2022 milestone to keep things moving forward, especially since August slipped by with very little forward progress!

**Comment by oleksandr-danylchenko** [comment]
Okay, after you provided the `gettext` reference the `_` started to make more sense to me. Could you please leave the `gettext` reference and a bit of explanation somewhere in the JSDoc for the `_(msg, context) {` method?

**Comment by oleksandr-danylchenko** [comment]
Okay, after you provided the `gettext` reference the `_` started to make more sense to me :ok_hand:.  
Could you please leave the `gettext` reference and a bit of explanation somewhere in the JSDoc for the `_(msg, context) {` method?

**Comment by JoeGermuska** [comment]
the `role` is different in the two locations; in `Timeline.js` (which is to be removed), the `role` is `group`, but in `TimeNav.js` the role is `application`.

The [MDN ARIA documentation](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles) lists that value under its "Avoid using" section.

@oleksandr-danylchenko [you have a comment](https://github.com/NUKnightLab/TimelineJS3/blob/accessibility-i18n/src/js/timenav/TimeNav.js/#L45), `// Prevents inconsistent default keyboard handling by the screen readers`

Does that refer to using the role "application"?  Did you try (or consider) the [navigation](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/navigation_role) role?

If I had a better accessibility testing rig set up I'd just try it myself, but I don't, and you've invested so much effort in this that I thought asking you was the best alternative.

**Comment by oleksandr-danylchenko** [comment]
> in `TimeNav.js the role is `application`

This is actually **a well-planned step** I did to prevent default browser handling of the `Tab` and arrow keys. I also discussed this with the a11y expert of our team. W/o it, in complex components, we can predictably control neither the keyboard navigation order nor the visual representi.

Quote:
> Screen readers have two modes generally. A direct input mode where the key presses are passed through to whatever program/app you are using
and
a key intercept mode in which the screen reader intercepts key presses for navigation purposes.
In the key intercept mode, most key presses will be intercepted on a webpage because they are generally expected to be consumed and lightly interacted with rather than being a full application."

> Using the Aria role application lets the user know there is some special interaction there that needs to function like an application rather than a web page.
Activating this area with enter or tabbing to it in JAWS and NVDA will activate their respective application modes which will turn key intercepts off. It's expected behavior.

Adding a proper screen reader behavior was especially painful for the `TimeNav` before I used the `application`. Example of the Chrome behavior before I appliance: 

https://user-images.githubusercontent.com/68850090/187986720-b6ff8cfe-161f-4697-9481-b0f44dfb18cd.mp4

But in Firefox, the elements didn't jump. In Opera they jumped exactly the same as in Chrome.
